android/src/com/mapswithme/maps/Framework.java

```diff
@@ -61,6 +61,13 @@
     public boolean buildings;
   }
 
+  public static class RouteAltitudeLimits
+  {
+    public int minRouteAltitude;
```

Comment:
> What's the problem to get current Route, and create a couple of methods, like:
GenerateChart(Route const & route, ...)
GetMinMaxAltitude(Route const & route, ...)

Saying methods, which class are you talking about? Framework? RoutingSession? Route? We can lock the current route only within RoutingSession methods. Or as an alternative we can copy instance or the current route (RoutingSession::m_route) and then work with it.